### DISCUSSION TITLE
Custom NGINX config generation

### DISCUSSION THREAD

**Discussion by ansidev** [nginx-proxy/nginx-proxy#1706] [Q&A]
Currently NGINX generated config:
```nginx
upstream server1-upstream {
	server 1.2.3.4:80;
}

server {
	server_name example.tld;
	location / {
		proxy_pass http://example.tld-upstream;
	}
}
```

How to generate NGINX config automatically like this:
```nginx
upstream example.tld-upstream {
	server 1.2.3.4:80;
}

upstream example.tld-slash-path1-upstream {
	server 1.2.3.5:80;
}

upstream example.tld-slash-path2-upstream {
	server 1.2.3.6:80;
}

server {
	server_name example.tld;
	location / {
		proxy_pass http://example.tld-upstream;
	}

	location ^~ /path1 {
		proxy_pass http://example.tld-slash-path1-upstream;
	}

	location ^~ /path2 {
		proxy_pass http://example.tld-slash-path2-upstream;
	}
}
```

Thanks!

**Comment by buchdag** [comment]
Please take a look at the [path based routing feature](https://github.com/nginx-proxy/nginx-proxy/tree/dev#path-based-routing) we just added to the `dev` branch / image, that might be what you're looking for.

This feature is still experimental and only available on the `nginxproxy/nginx-proxy:dev` and `nginxproxy/nginx-proxy:dev-alpine` images.
